design/back.compatibility.md

```diff
@@ -0,0 +1,79 @@
+## Back Compatibility Plan
+For MRE, back compatibility is an important promise to all stakeholders of the MRE ecosystem. This proposed change will allow MRE client plugins to support multiple "generations" of the MRE SDK. Each generation may have functionality additions or changes in both API surface and protocol, and once a generation goes from preview to finalized release, it's locked for good, and a new generation enters preview phase.
```

Comment:
>once a generation goes from preview to finalized release, it's locked for good, and a new generation enters preview phase [](start = 293, length = 121)

So in the context of this first release, v0 is locked and v1 enters preview?

Reply:
I think that SDK 1.1, 1.2, etc should map to the gen1 runtime. Point releases of the SDK should improve stability, not introduce future instability. SDK v2.0.0-prerelease.0 should map to gen2, but may need to be updated when the gen2 runtime is finalized.

Reply:
Changing this slightly, so that 1.n is the final version for generation 1, and 2.0 is first preview of gen 2. That way, it is possible to do hotfixes (1.n+1 etc.) if needed (though as noted elsewhere, hotfixes should be done in a very limited fashion.